pkg/cli/verifier/verifier.go

```diff
@@ -0,0 +1,93 @@
+package verifier
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/fatih/color"
+)
+
+// Status is a type describing the status of a verification
+type Status string
+
+const (
+	// Success indicates the verification succeeded
+	Success Status = "Success"
+
+	// Failure indicates the verification failed
+	Failure Status = "Failure"
+
+	// Unknown indicates the result of the verification could not be determined
+	Unknown Status = "Unknown"
+)
+
+// Result defines the result returned by a Verifier instance
+type Result struct {
+	Context       string
+	Status        Status
+	Reason        string
+	Suggestion    string
+	NestedResults []*Result
```

Comment:
I assume this is a slice of pointers to reduce excessive copying?

Reply:
Yes, although this could very well have been a copy given the struct size is small.